pkg/kubelet/util/pluginwatcher/plugin_watcher_test.go

```diff
@@ -154,27 +75,20 @@ func TestExamplePlugin(t *testing.T) {
 	// grpcServer.Stop() will remove the socket and starting plugin service
 	// at the same path again should succeeds and trigger another callback.
 	require.NoError(t, p.Stop())
-	p = NewTestExamplePlugin("")
-	go func() {
-		require.Nil(t, p.Serve(socketPath))
-	}()
-	require.True(t, waitForPluginRegistrationStatus(t, p.registrationStatus))
+	require.Nil(t, p.Serve(socketPath))
+	require.False(t, waitForPluginRegistrationStatus(t, p.registrationStatus))
```

Comment:
Why successful registration is not expected in this case?

Reply:
this is two questions, 
to answer the first question, the example handler has been modified to know which plugin name already registered, so when a plugin been restarted (watcher does not), it try to register again with same name, it will getting an error from handler.

could you clarify a little bit for the second question:
the plugin gRPC server is always there until p.Stop() been called,
so `waitForPluginRegistrationStatus` should work fine, my test does shows working fine, thanks

Reply:
> to answer the first question, the example handler has been modified to know which plugin name already registered, so when a plugin been restarted (watcher does not), it try to register again with same name, it will getting an error from handler.

@figo at L#77 plugin is stopped. Should not this unregister it? and if yes, registration at L79 should not fail? WDYT?

Reply:
@vikaschoudhary16 thanks for take a good again, please correct me if my understand is wrong since this change is based on your change, 

PluginWatcher and handler does not handle unregistration at this moment, i saw you are actively discuss it at #64637,  before unregistration been implemented, L77 will not trigger anything.

and this change introduced a change on example handler:  each handler will remember registered plugin names,

when L79 trigger a registration process, 
handler will return error of "plugin name already exist".

once unregistration been implemented, this test do need be updated.

Reply:
/lgtm
make sense. Thanks!